app/Traits/CallsBucketsTrait.php

```diff
@@ -38,6 +38,43 @@ public function authorizeAWS($source)
             return $security_token;
         }
     }
+    /**
+     * Method to sign a URL but the CloudFrontClient object should be passed by parameter
+     * to do an only one call to AWS
+     *
+     * @param CloudFrontClient $client
+     * @param string $file_path
+     * @param int $transaction
+     *
+     * @return string signed url
+     */
+    public function signedUrlUsingClient(CloudFrontClient $client, string $file_path, int $transaction): string
+    {
+        if (!$client) {
+            return null;
+        }
+
+        $cdn_server_url = config('services.cdn.server');
+        $request_array = [
+            'url'         => 'https://'. $cdn_server_url . '/' . $file_path . '?x-amz-transaction=' . $transaction,
+            'key_pair_id' => config('filesystems.disks.cloudfront.key'),
+            'private_key' => storage_path('app/' . config('filesystems.disks.cloudfront.secret')),
+            'expires'     => Carbon::now()->addDay()->timestamp,
+        ];
+
+        $signed_url = $client->getSignedUrl($request_array);
+        $query_parameters_string = parse_url($signed_url, PHP_URL_QUERY);
+        $query_parameters = [];
+        parse_str($query_parameters_string, $query_parameters);
+        $key = $this->getKey();
+
+        if (!empty($key)) {
+            $signature = isset($query_parameters['Signature']) ? $query_parameters['Signature'] : $signed_url;
+            \Log::channel('cloudfront_api_key')->notice($key . ' ' . $signature);
+        }
+
+        return $signed_url;
+    }
 
     public function signedUrl(string $file_path, $asset_id, int $transaction)
     {
```

Comment:
now with the above signedUrlWithClient function, please refactor signedUrl to authenticate and call signedUrlWithClient. Also, since we no longer serve content from S3, please remove the code starting with the comment  // Or return S3 Urls

Reply:
Ok @bradflood, thank you. So, I am going to do the refactoring for the signedUrl method.

Reply:
@bradflood The refactor for the signedUrl method has been done.